kwok.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Jan  3 15:44:42 2020
+
+@author: hanan
+"""
+
+__doc__ = """
+
+Dependencies used (subject to change): pandas, numpy, scipy.stats
+
+This .py file is meant to handle these 4 features/variables in feature engineering:
+    GarageArea - the size/area of the garage in square feet
+    GarageCars - the car capacity of the garage (number of cars)
+    OpenPorchSF - the size/area of the open porch in square feet
+    WoodDeckSF - the size/area of the wood deck in square feet
+    
+The function named feature_extract is where the dataset will be passed in & the 4 features
+will be extracted for processing, including handling of missing values & outliers
+and possibly normalization/handling of skewed data.
+"""
+
+import pandas as pd # import pandas in order to be able to bring in the dataset
+import numpy as np # import numpy to be able to take the log transformation
+from scipy.stats import skew # import skew to be able to handle skew 
+
+def feature_extract(dataset:pd.DataFrame) -> pd.DataFrame:
+    """
+    A function for the extraction of these 4 variables/features:
+        GarageArea
+        GarageCars
+        OpenPorchSF
+        WoodDeckSF
+        
+    Parameters
+    ----------
+    dataset : pd.DataFrame
+        This is the dataset that will be passed into the function (as a parameter) 
+        for processing.
+
+    Returns
+    -------
+    myFeatures : pd.DataFrame
+        This is the DataFrame containing the 4 selected features (mentioned 
+        above) that will be returned as output.
+
+    """
+    
+    # Picking out the assigned columns & putting them into a separate DataFrame
+    myFeatures = pd.DataFrame(dataset, columns = ['GarageArea', 'GarageCars', 'OpenPorchSF', 'WoodDeckSF'])

```

Comment:
`myFeatures = dataset[['GarageArea', 'GarageCars', 'OpenPorchSF', 'WoodDeckSF']]`
This would be a more efficient way to grab the columns you're specifically looking for. You don't want to get into the habit of creating variables that take extra memory especially working with large data like we are. Keep in mind this is just me looking at the details, but these details are things that really matter on a large business scale.

Reply:
In fact, as written, it would throw an error. Consider changing to Alex's suggestion.

Reply:
I updated the code just now to have the suggested syntax.